packages/flutter_tools/gradle/aar_init_script.gradle

```diff
@@ -0,0 +1,116 @@
+import org.gradle.api.Project
+import org.gradle.api.artifacts.Configuration
+import org.gradle.api.artifacts.maven.MavenDeployer
+import org.gradle.api.plugins.MavenPlugin
+import org.gradle.api.tasks.Upload
+
+void configureProject(Project project, String buildDir) {
+    if (!project.hasProperty("android")) {
+        throw new GradleException("Android property not found.")
+    }
+    if (!project.android.hasProperty("libraryVariants")) {
+        throw new GradleException("Can't generate AAR on a non Android library project.");
+    }
+
+    project.apply plugin: "maven"
+
+    project.android.libraryVariants.all { variant ->
+        setupLibraryVariant(project, buildDir, variant)
+    }
+
+    project.uploadArchives {
+        repositories {
+            mavenDeployer {
+                repository(url: "file://${buildDir}/repo")
+            }
+        }
+    }
+
+    Boolean usesFlutterPlugin = project.plugins.find { it.class.name == "FlutterPlugin" } != null
+    if (!usesFlutterPlugin) {
+        // The Flutter plugin already adds `flutter.jar`.
+        project.dependencies {
+            compileOnly project.files("${getFlutterRoot(project)}/bin/cache/artifacts/engine/android-arm-release/flutter.jar")
+        }
+    }
+}
+
+String getFlutterRoot(Project project) {
+    if (!project.hasProperty("flutter-root")) {
+        throw new GradleException("The `-Pflutter-root` flag must be specified.")
+    }
+    return project.property("flutter-root")
+}
+
+void setupLibraryVariant(Project project, String buildDir, variant) {
+    String variantName = variant.name.capitalize()
+    Task aarTask = project.tasks.create(name: "aar${variantName}") {
```

Comment:
Gradle now recommends to use tasks.register rather than tasks.create so configuration can be lazy:

https://docs.gradle.org/current/userguide/task_configuration_avoidance.html

Reply:
Thanks for the feedback! I added this to https://github.com/flutter/flutter/issues/33064